### ISSUE TITLE
Default RolePolicy overriding my own RolePolicy

### ISSUE THREAD

**Issue by hackel** [vyuldashev/nova-permission#45]
The default "allow all" RolePolicy included in this package in 7a1fdfd is overriding my own RolePolicy, and allowing unprivileged users to assign roles they aren't authorized to assign.  I assume the same issue exist for PermissionPolicy, though I haven't tested that.

One way to fix it is to simply move the gate registrations inside the empty Nova::serving call in the oddly named ToolServiceProvider.  This is still limiting, however, as even in Nova one may wish to customize the policy and prevent certain roles/permissions from being assigned, removed, etc.

I'm not clear on why these policies exist at all?  My understanding is that, in the absence of a policy, by default Nova will allow all actions on a resource.  It seems like it would be much better to simply remove the policies and allow users to create their own if they require them.

**Comment by vyuldashev** [comment]
> I'm not clear on why these policies exist at all? My understanding is that, in the absence of a policy, by default Nova will allow all actions on a resource. It seems like it would be much better to simply remove the policies and allow users to create their own if they require them.

We added them because of navigation. Are you using custom `Role` class instead of spatie's ones from the package?

**Comment by hackel** [comment]
@vyuldashev Yes I am, but the policy is registered for the `Spatie\Permission\Contracts\Role` interface which resolves to my custom App\Role class.  I see now that yours is registered for the actual class specified in the config file.  If I change mine to register the policy for my class directly, yours no longer overrides it.

Relying on the interface seems like it's probably the more "correct" approach to me, but it's debatable.  In the meantime, I can use this work-around.

What needs to be present in my policy for navigation to work correctly?  Just "viewAny"?

**Comment by vyuldashev** [comment]
I will look into it.

> What needs to be present in my policy for navigation to work correctly? Just "viewAny"?

There are more methods available: https://github.com/vyuldashev/nova-permission/blob/master/src/PermissionPolicy.php

**Comment by parasume** [comment]
I was working on this issue to submit a PR but it had a lot of breaking changes and it actually ended up being a re-write of this package.

So, I've created a new package: https://github.com/insenseanalytics/laravel-nova-permission

Key differences are as follows:
1. Solves this issue and #4 
2. Also added the ability to use permissions based authorization for Nova resources (instead of policy based) if you'd like to do that for certain resources

Both forgetting the cache on attaching permissions and permission based authorization were critical for my application since we link permissions with roles (rather than directly). Feel free to use it if you're in a similar boat. Cheers!
